site/content/blog/2021/2021-11-02-ADRs-Das-Format.adoc

```diff
@@ -0,0 +1,88 @@
+Dabei gehe ich insbesondere auf den Kontext des Dokumentationstemplates arc42 und dessen Einsatz im DevOps-Umfeld ein.
+
+Wir sehen uns dazu auch intensiver die Technik an, mit der dokumentiert wird. Hier kommen AsciiDoc, Microsites und Git zum Einsatz.
+
+Aber starten wir zuerst mit dem _Warum_. Warum ist es eine gute Idee, Entscheidungen zu dokumentieren?
+
+== Undokumentierte Entscheidungen - Ein Problem auf unterschiedlichen Ebenen
+
+Undokumentierte Entscheidungen sind oft die Regel. Meistens bleiben sie eine lange Zeit unbemerkt oder kommen nie wieder zur Sprache.
+Interessant wird es, wenn an einer Stelle die Frage gestellt wird: *Warum* ist das *so*?
+
+Dabei ist dieses Problem nicht nur auf technische Dinge beschränkt. Auch zwischenmenschliche Aspekte der Zusammenarbeit sind betroffen.
+
+=== Technische Entscheidungen
+
+Technik kann alles Mögliche umfassen. Ich nehme mal das Beispiel einer Frameworkentscheidung für einen Microservice heraus.
+
+Inzwischen gibt es zu jeder Programmiersprache unzählige Frameworks, um einen Microservice zu implementieren. Interessanterweise gibt es in einem Team
+normalerweise mehrere Meinungen, welches Framework das Beste ist. Deswegen sollte so eine Entscheidung dokumentiert werden. Wichtig hierbei aus meiner Sicht: Die 
+untersuchten Alternativen mitdokumentieren.
+
+In unserem Team hatten wir genau diese Entscheidung zur Wiedervorlage. Wir waren mit unserer alten Implementierung unzufrieden und haben uns einige Alternativen
+angesehen und sie mit Performance-Tests und Proof-Of-Concepts evaluiert. Die folgende Entscheidung beruhte auf der erstellten Datenbasis. Alle Alternativen
+sind in der Entscheidungsdokumentation zu finden und für jeden nachvollziehbar.
+
+=== Zwischenmenschliches
+
+Zwischenmenschliches klingt meistens banal, hat aber im Alltag große Auswirkungen auf Arbeitsmoral und Motivation.
+Dazu ein Beispiel:
+
+In meinem Team fehlt anscheinend eine Regel, wie mit temporärer Abwesenheit ohne Krankmeldung umgegangen wird. Ein Teamkollege von mir war nur zum Daily
+anwesend, aber dann den ganzen Tag nicht mehr in der Teamkommunikation zu sehen. Am Ende des Tages war ich besorgt und leicht ärgerlich zugleich.
+Mir war nicht klar, wie er den Tag verbracht hat oder ob etwas passiert ist.
+
+Aus meiner Sicht brauchen wir dafür im Team eine dokumentierte Entscheidung, wie wir mit solchen Abwesenheiten umgehen.
+Ziel ist nicht Kollegen:innen zu gängeln, wenn sie sich nicht daran halten. Aber ich kann dann so darauf hinweisen: "Sieh mal, so hatten wir das entschieden,
+warum machst Du das nicht so?"
+
+== Architekturentscheidungen
+
+Eine besondere Rolle nehmen Entscheidungen ein, die sich auf die Softwarearchitektur eines Softwaresystems beziehen. Meine Definition dieser Entscheidungen ist
+die Folgende: "Eine Architekturentscheidung ist im Nachhinein potentiell teuer bis gar nicht zu ändern und bezieht sich auf die Qualitätsanforderungen eines Systems"
+
```

Comment:
Des weiteren wird Architektur z.B. als "die Summe der Entscheidungen, die, wenn sie falsch getroffen werden, das Projekt scheitern lassen" definiert.
Das zeigt schon stark den Stellenwert der Entscheidungen in einer Architektur.